### PR TITLE
Editing pass on spam

### PR DESCRIPTION
**Pull request by henryre**
Moved lots to utils, etc. Still needs some love, but off my stack.

### CODE REVIEW

File: spam/01_spam_tutorial.py [hunk 1]
```diff
@@ -99,23 +99,29 @@
 # * The fifth video is split 50/50 between a validation set (`valid`) and `test` set.
 
 # %% [markdown]
-# This next cell takes care of some notebook-specific housekeeping.
+# This next two cell takes care of some notebook-specific housekeeping.
```
Comment (brahmaneya) [inline spam/01_spam_tutorial.py]: cells take

File: spam/01_spam_tutorial.py [hunk 2]
```diff
@@ -892,33 +872,17 @@ def plot_probabilities_histogram(Y):
...
-)
-optimizer = tf.keras.optimizers.Adam(lr=0.01)
-keras_model.compile(
-    optimizer=optimizer, loss="categorical_crossentropy", metrics=["accuracy"]
-)
-
-early_stopping = tf.keras.callbacks.EarlyStopping(
-    monitor="val_acc", patience=10, verbose=1, restore_best_weights=True
-)
+keras_model = get_keras_logreg(input_dim=X_train.shape[1])
```
Comment (brahmaneya) [inline spam/01_spam_tutorial.py]: What is logreg? I'd just call this get_keras_model or get_keras_mlp

File: spam/02_spam_data_augmentation_tutorial.py [hunk 3]
```diff
@@ -16,40 +16,45 @@
...
-# ## 1. Loading Data
-
-# %% [markdown]
-# We load the Kaggle dataset and create Pandas DataFrame objects for each of the sets described above.
-# The two main columns in the DataFrames are:
-# * **`text`**: Raw text content of the comment
-# * **`label`**: Whether the comment is `SPAM` (1) or `HAM` (0).
-#
-# For more details, check out the labeling functions [tutorial](https://github.com/snorkel-team/snorkel-tutorials/blob/master/spam/01_spam_tutorial.ipynb).
+# This next two cell takes care of some notebook-specific housekeeping.
```
Comment (brahmaneya) [inline spam/02_spam_data_augmentation_tutorial.py]: cells take

File: spam/02_spam_data_augmentation_tutorial.py [hunk 4]
```diff
@@ -16,40 +16,45 @@
...
+# You can ignore it.
```
Comment (brahmaneya) [inline spam/02_spam_data_augmentation_tutorial.py]: ignore them

File: spam/01_spam_tutorial.py [hunk 5]
```diff
@@ -99,23 +99,29 @@
 # * The fifth video is split 50/50 between a validation set (`valid`) and `test` set.
 
 # %% [markdown]
-# This next cell takes care of some notebook-specific housekeeping.
+# This next two cell takes care of some notebook-specific housekeeping.
 # You can ignore it.
```
Comment (brahmaneya) [inline spam/01_spam_tutorial.py]: ignore them

File: spam/02_spam_data_augmentation_tutorial.py [hunk 6]
```diff
@@ -16,40 +16,45 @@
...
-    os.chdir("spam")
+# %%
+# Download the spaCy english model
+# ! python -m spacy download en_core_web_sm
+
+# %% [markdown]
+# ## 1. Loading Data
+
+# %% [markdown]
+# We load the Kaggle dataset and create Pandas DataFrame objects for each of the sets described above.
```
Comment (brahmaneya) [inline spam/02_spam_data_augmentation_tutorial.py]: objects for the train, test and validation sets (There are no sets described above anymore.)

File: spam/02_spam_data_augmentation_tutorial.py [hunk 7]
```diff
@@ -271,16 +287,18 @@ def replace_adjective_with_synonym(x):
 print(f"Augmented training set size: {len(df_train_augmented)}")
 
 # %% [markdown]
-# We have almost doubled our dataset using TFs! Note that despite `n_per_original` being set to 2, our dataset may not exactly triple in size, because sometimes TFs return `None` instead of a new example (e.g. `change_person` when applied to a sentence with no persons).
+# We have almost doubled our dataset using TFs!
+# Note that despite `n_per_original` being set to 2, our dataset may not exactly triple in size, because sometimes TFs return `None` instead of a new example (e.g. `change_person` when applied to a sentence with no persons).
```
Comment (ajratner) [inline spam/02_spam_data_augmentation_tutorial.py]: This is actually an interesting point... should we be making sure that each data point gets copied exactly the same number of times?  I think this is a more proper way of doing this... should discuss IRL

File: spam/02_spam_data_augmentation_tutorial.py [hunk 8]
```diff
@@ -291,59 +309,51 @@ def replace_adjective_with_synonym(x):
...
+def train_and_test(
+    X_train,
+    Y_train,
+    X_valid=X_valid,
+    Y_valid=Y_valid,
+    X_test=X_test,
+    Y_test=Y_test,
+    num_buckets=30000,
+):
+    lstm_model = get_keras_lstm(num_buckets)
```
Comment (ajratner) [inline spam/02_spam_data_augmentation_tutorial.py]: I think this is great but should include a comment saying something to the effect of 'this is *just* a vanilla LSTM in Keras that we put in the util file for cleanliness'

File: spam/01_spam_tutorial.py (hunk 2 above)
Comment (ajratner) [reply spam/01_spam_tutorial.py]: (@brahmaneya Logistic regression; I think that abbreviation standard enough to leave in)

@henryre I think this is good to put this stuff in util file but still need to call out that it's *just* standard keras implementation (eg add comment)

File: spam/02_spam_data_augmentation_tutorial.py [hunk 9]
```diff
@@ -1,55 +1,81 @@
...
-# For an overview of Snorkel, visit [snorkel.org](http://snorkel.org).
-# You can also check out the [Snorkel API documentation](https://snorkel.readthedocs.io/).
+# In this tutorial, we will walk through the process of using transformation functions (TFs) to perform data augmentation.
+# Like the labeling tutorial, our goal is to train a classifier to YouTube comments as `SPAM` or `HAM` (not spam).
+# While data augmentation can be used in concert with labeling functions,
+# we assume that we have access to some labeled YouTube comments for training for the purposes of this tutorial.
 #
-# For our task, we have access to some labeled YouTube comments for training. We generate additional data by transforming the labeled comments using **_Transformation Functions_**.
+# * For more details on the task, check out the [labeling tutorial](https://github.com/snorkel-team/snorkel-tutorials/blob/master/spam/01_spam_tutorial.ipynb)
+# * For an overview of Snorkel, visit [snorkel.org](http://snorkel.org)
```
[the remark this answers is not in the document]
Comment (henryre) [reply spam/02_spam_data_augmentation_tutorial.py]: i'll add as a separate cell that doesn't render output

**Pull request merged**